src/traces/parcoords/parcoords.js

```diff
@@ -450,7 +508,110 @@ module.exports = function(root, svg, parcoordsLineLayers, styledData, layout, ca
+        // reset hover properties for all axes, save it only for currently hovered axis
+        gd.data[0].dimensions = gd.data[0].dimensions.map((item) => {item.hover = null; return item});
+        gd.data[0].dimensions[hoveredAxisIndex].hover = setIndex;
+        Plotly.redraw(gd);
+    }
+
+    if (/*layout.showProbabilityDensity && */ layout.showProbabilityDensity != 'never') {
+        var showDensityOnHover = layout.showProbabilityDensity === 'hover';
+
+        var densityGroupJoin = yAxis.selectAll('g.density')
+            .data(function (d) {
+                if (showDensityOnHover && !d.hover) {
+                    return [];
+                } else {
+                    return [d];
+                }
+            });
+
+        var densityGroup = densityGroupJoin.enter()
+            .append('g')
+            .classed('density', true);
+
+        // TODO: Can't use linePoints like in boxPlot, as in parcoords, there is no plotinfo for axes, which defines c2p, properties and other
+        var densitySegments = function (d, sideFlip) {
+
+            var outerArray = new Array(d.probabilityDensity.density.length),
+                scaleMax = Math.max(...d.probabilityDensity.scale),
+                densityMax = Math.max(...d.probabilityDensity.density),
+                scaleFactor = d.height / scaleMax,
+                densityFactor = d.model.canvasWidth / d.model.colCount / densityMax * 0.5;
+            for (var i = 0; i < outerArray.length; i++) {
+                var innerArray = new Array(2);
+                innerArray[0] = d.probabilityDensity.density[i] * densityFactor * sideFlip;
+                innerArray[1] = d.probabilityDensity.scale[i] * scaleFactor;
+                outerArray[i] = innerArray;
+            }
+            return outerArray;
+        };
+
+        var drawDensity = function(leftSide) {
+            var sideClass = leftSide ? "left" : "right";
+
+            densityGroup
+                .selectAll('path.' + sideClass)
+                .data(function(d) {
+                    return [densitySegments(d, leftSide ? -1 : 1)]
+                })
+                .enter().append('path')
+                .classed('js-line ' + sideClass, true)
+                .style('vector-effect', 'non-scaling-stroke')
+                .call(Drawing.lineGroupStyle, 2, 'blue', '')
+                .each(function(d) {
+                    var smoothing = 1,
+                        path = Drawing.smoothopen(d, smoothing),
+                        xScaleFactor = 20
+                    d3.select(this)
+                        .attr('d', path)
+                        // .call(Drawing.lineGroupStyle);
+                });
+        };
+
+        drawDensity(true);
+        drawDensity(false);
+
+        densityGroupJoin.exit()
+            .remove();
+
+        if (showDensityOnHover === true) {
+            yAxis.on('mouseover', function(/*datum, index, currentThis*/) {
```

Comment:
Rekl bych, ze se listener registruje nekolikrat a nikdy se nederegistruje. Chtelo by to pred tim volat `.off()`